File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        int c = 0;
        return kthSmallest(root, k, c);
    }

    int kthSmallest(TreeNode* root, int k, int& c) {
        if(root == nullptr) return -1;
        int val = kthSmallest(root -> left, k, c);
        if(val != -1) return val;
        ++c;
        if(k == c) return root -> val;
        return kthSmallest(root -> right, k, c);
    }

};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        

        def dfs(root, greatestVal) -> int:
            res = 0
            if root is None:
                return 0
            if(root.val >= greatestVal):
                res += 1
                greatestVal = root.val
            
            res += dfs(root.left, greatestVal)
            res += dfs(root.right, greatestVal)

            return res
        
        return dfs(root, root.val)
        